exercises/multiple_promises/solution/solution.js

```diff
@@ -1,31 +1,33 @@
-'use strict';
+function all(promise1, promise2) {
+    var fulfill, reject;
 
-/* global getPromise1, getPromise2 */
+    var p = new Promise((ff, rj) => {
+        fulfill = ff;
+        reject = rj;
+    });
 
-function all(a, b) {
-  return new Promise(function (fulfill, reject) {
     var counter = 0;
     var out = [];
 
-    a.then(function (val) {
-      out[0] = val;
-      counter++;
+    promise1.then((value) => {
+        counter++;
+        arr.push(value);
```

Comment:
By using `array.push(value)`, `out` will contain values in order of when the promises are called. So lets say, `promise1` is called earlier than `promise2`, the result will be `[value1, value2]`, but if `promise2` is called earlier the result will be `[value2, value1]`. Wouldn't it be more convenient if the order of the values of `out` was corresponding to the arguments order of `all`?